e2e/nomostest/testutils/validate.go

```diff
@@ -40,3 +41,15 @@ func ValidateError(errs []v1beta1.ConfigSyncError, code, message string) error {
 	}
 	return errors.Errorf("error %s not present: %s", code, log.AsJSON(errs))
 }
+
+// WithFinalizer adds a finalizer to the object
+func WithFinalizer(obj client.Object, finalizer string) {
```

Comment:
this With- pattern is more for builder options... perhaps AppendFinalizer & RemoveFinalizers/ClearFinalizers/DeleteFinalizers?

Reply:
done